pkg/Cpanel/Security/Advisor/Assessors/Mysql.pm

```diff
@@ -109,6 +113,32 @@ sub _check_for_anonymous_users {
     return 1;
 }
 
+sub _check_for_public_bind_address {
+    my $self = shift;
+
+    my $mycnf        = Cpanel::MysqlUtils::MyCnf::Full::etc_my_cnf();
+    my $bind_address = $mycnf->{'mysqld'}->{'bind-address'};
+
+    my @deny_rules   = grep { /--dport 3306/ && /-j (DROP|REJECT)/ } split /\n/, Cpanel::SafeRun::Errors::saferunnoerror( '/sbin/iptables',  '--list-rules' );
+    my @deny_rules_6 = grep { /--dport 3306/ && /-j (DROP|REJECT)/ } split /\n/, Cpanel::SafeRun::Errors::saferunnoerror( '/sbin/ip6tables', '--list-rules' );
+
+    # This is a very basic check.
+    # We are looking to see if the bind address unset or set to anything other than localhost.
+    # We also check to see if there are any likely deny/reject rules on the SQL port.
+
+    if ( ( defined($bind_address) && Cpanel::IP::Loopback::is_loopback($bind_address) ) || @deny_rules || @deny_rules_6 ) {
+        $self->add_good_advice( text => "MySQL is listening only on a local address." );
```

Comment:
This statement is false if the user relies on deny rules.  I would split that out into a separate section for more accuracy, something like:

```perl
if (...::is_loopback($bind_address)) {
    $self->add_good_advice(...);
}
elsif (@deny_rules ...) {
    $self->add_info_advice( text=> "The MySQL port is blocked by the firewall, effectively allowing only local connections");
}
```